### PR TITLE
[Docs] Guide for practicing schema driven development for Meshery Server.

### PR DESCRIPTION
**Pull request by MUzairS15**
**Notes for Reviewers**

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit f5c24c9342ba0eae58f77e82a138851a1a5ab6a5 at: https://66ace3e06e8dc35a49410a1e--meshery-docs-preview.netlify.app

File: docs/pages/project/contributing/meshery-server.md [hunk 1]
```diff
@@ -104,3 +104,134 @@ github.com/layer5io/meshkit => ../meshkit
   </li>
 </ol>
 
+### Practicing Schema driven development for Meshery server.
+The guide covers instructions on how to generate Go code from OpenAPI specifications, best practices to ensure the generated code meets the requirements, follows the conventions of the project and necessary pre/post-requisites.
+
+### Installating required tools
+1. `oapi-codegen`: https://github.com/oapi-codegen/oapi-codegen?tab=readme-ov-file#install
```
Comment (leecalcote) [inline docs/pages/project/contributing/meshery-server.md:111]: @MUzairS15 describe each of these tools.

Comment (leecalcote) [reply docs/pages/project/contributing/meshery-server.md:111]: @MUzairS15 combine the "Install Required Tools" with the "Prerequisites" section below. This should be a step within that section.

File: docs/pages/project/contributing/meshery-server.md [hunk 2]
```diff
@@ -104,3 +104,134 @@ github.com/layer5io/meshkit => ../meshkit
...
+2. `redocly`: `npm i -g @redocly/cli`
+
+## Prerequisite
+1. Ensure you have forked the [meshery/schemas](github.com/meshery/schemas) and working directory is `schemas` repository.
+2. Resolving References in JSON Schemas:
```
Comment (leecalcote) [inline docs/pages/project/contributing/meshery-server.md:116]: This step is out of place; not well-described. There's no lead-in or explanation.

File: docs/pages/project/contributing/meshery-server.md [hunk 3]
```diff
@@ -104,3 +104,134 @@ github.com/layer5io/meshkit => ../meshkit
...
+    The OpenAPI spec references other JSON schemas, which in turn has references to other schemas, the `oapi-codegen` tool fails to resolve such nested references, hence the references needs to be resolved before generating the code.
+    Use the `ref-resolver.js` script provided in the `schemas` (scripts/ref-resover.js) repository to perform the resolution:
+
+_If the openapi spec points to a ref, eg: ../a.json and the schema inside a.json doesn't have any other references then the `oapi-codegen` works fine, the task of the `ref-resovler.js` is to just ensure the nested external references are resolved to the point `oapi-codegen` can work._
+
+`Run: make resolve-ref path=<file/directory containing the schemas whose references needs to be resolved (the path should be relative from the root of the repository)> eg: make resolve-ref path=schemas/constructs/v1beta1.`
+
+## Generating constructs
+```bash
```
Comment (leecalcote) [inline docs/pages/project/contributing/meshery-server.md:125]: Step 3?

**Review by leecalcote — CHANGES_REQUESTED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit 8289d18d8c313cada92e69b99f8649fe7c797b2d at: https://66b1295e957417041a71eedd--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
Willfix

**Pull request closed without merging**